### DISCUSSION TITLE
[Standalone] Ninja Assault: Test Mode and Mouse Config (NAOMI)

### DISCUSSION THREAD

**Discussion by kathyrollo** [flyinghead/flycast#706] [General]
Platform / OS: Win10 64bit

Github hash: 2022-07-22T16:15:23.000Z ([5ae7ac3](https://github.com/flyinghead/flycast/commit/5ae7ac3))

Hardware: R7 4800H / GTX 1650 Ti / 16GB RAM

**_Description of the Issue_**
Hi @flyinghead ,
I can enter Test Mode with the key I mapped but unable to select through the options using the Switch key. This appears to only happen in Namco games like Ninja Assault and Gun Survivor 2 Code Veronica. The Service key works with The House of the Dead 2 and Lupin The Shooting (or all other non-Namco games).

**_Debugging Steps Tested_**

1. Press F2 to enter Test Mode.
2. Press F1 to cycle through options.

Nothing happens. The first option just blinks red and white.

**_Screenshots_**
Keyboard (Global):
![Screenshot_1](https://user-images.githubusercontent.com/28589393/180645648-21ca3780-a24d-4be9-9a77-aaea1c70e3d9.jpg)

Test Mode (Ninja Assault):
![namco_test_mode](https://user-images.githubusercontent.com/28589393/180646104-cc5e960e-842f-41de-8ff4-cbf8de48bf63.gif)

**Comment by flyinghead** [comment]
Press Up and Down to select the entry and Start to enter.

**Comment by kathyrollo** [comment]
Hi @flyinghead , was able to move up and down the options in the "Menu" above with arrow keys and discovered I need to press F2 during the "NAOMI" bootscreen to enter the actual "System Menu":

ninjasltu.zip (non-merged 0.245):
![20220725_133334](https://user-images.githubusercontent.com/28589393/180695042-0d47c15b-f6a2-4ff0-ae63-aa42efed4405.jpg)

![20220725_133455](https://user-images.githubusercontent.com/28589393/180695061-3bb7d99e-cc6f-49b4-9bc0-62750e2af6da.jpg)

I'm now able to cycle through options with Service (F1) and can enter sub-options like System Assignments with Test (F2) as usual. However, when I attempt Game Test Mode, it just reboots to the game. Any idea why?

**Reply by flyinghead** [reply]
This is the way this game works: you can't access the game test menu from the BIOS service menu. You need to press Test when the game is started.

Note: I moved this to Discussions. Please don't open issues for questions about how to use the Naomi platform or games.

**Reply by kathyrollo** [reply]
Thanks @flyinghead , was unaware of the Discussions feature of GitHub and actually looking for a forum-like board.

ninjasltu.zip (non-merged 0.245):
![20220725_191052](https://user-images.githubusercontent.com/28589393/180745246-f20f14c5-eaea-4bb9-861d-b3d3a4e1e075.png)

Back to the in-game Menu and able to move up/down via arrow keys as advised. Game Options now highlighted, but somehow cannot enter any sub-menu. Tried pressing the Return (Enter) key and nothing happens. Tried the Test (F2) key and it reboots to the game.

Keyboard (Global):
![Screenshot_2](https://user-images.githubusercontent.com/28589393/180745620-1e60e13c-aa6a-41e4-9b42-d1811f8c0612.jpg)

I play with Xbox One controller and mouse fulltime so I usually map the keyboard to "arcade machine" or emulator-specific functions only. Bit stumped if there's anything missing and why I can't enter Game Options?

**Reply by flyinghead** [reply]
Use Button 3 as indicated in the mapping window:
![image](https://user-images.githubusercontent.com/16261910/180747526-4403947c-8fa1-4e37-b855-3835712cf7ad.png)

**Comment by kathyrollo** [comment]
That did it, thanks @flyinghead! I have to admit this one has a lot of gotcha's. I was finally able to get crosshairs, calibrate gun (I/O Test), left-click (trigger), right-click (reload). HOTD2 was pretty straightforward from the global settings, while this had to be game-specific for keyboard and mouse. Posting what I ended up with for documentation.

Per Game Settings (or Global) - Controls:
![na_controls](https://user-images.githubusercontent.com/28589393/180761728-ea6606c4-4cd3-41b9-a59a-29c51d6e2c7c.jpg)

Per Game Settings - Keyboard:
![na_kb](https://user-images.githubusercontent.com/28589393/180761804-482262d5-98cf-4531-a016-2d6edcb1c9d1.jpg)

Per Game Settings - Default Mouse:
![na_mouse](https://user-images.githubusercontent.com/28589393/180762052-3b9bf2b2-dadd-4c51-b816-7e4817954f14.jpg)

Thanks again for the support.

**Comment by kathyrollo** [comment]
Attaching the official manual for this game. Navigation of the in-game menu and calibration is also here.
[Ninja Assault - Manual (SD)(U).pdf](https://github.com/flyinghead/flycast/files/9188804/Ninja.Assault.-.Manual.SD.U.pdf)

**Comment by kathyrollo** [comment]
Hi @flyinghead , a commenter pointed out in this timestamp there is white flashing on the pillarboxes while shooting. Would you know how to remove it?

https://youtu.be/ozQ8JJmf-8o&t=7m36s

**Reply by flyinghead** [reply]
This is a feature of the game: it changes the screen border color every time a shot is fired. This shows on the pillarboxes when they are visible.

One way to avoid this is to have a 4:3 screen/window. Or to enable widescreen/super widescreen.

**Reply by kathyrollo** [reply]
Thanks, will let them know. :)
